.circleci/config.yml

```diff
@@ -28,115 +52,121 @@ jobs:
+      - run:
+          name: Check Version matches Tag
+          command: make ci-check-version
+      - run:
+          name: Build and Publish Docker Images
           command: |
             docker login -u $DOCKER_USER -p $DOCKER_PASS
-            version=$(make version)
-            docker push vaporio/emulator-plugin:latest
-            docker push vaporio/emulator-plugin:$version
+            ./bin/ci/build_and_publish.sh latest
 
-  # release creates a GitHub release draft for a tag
+  # release
+  #
+  # This job creates a GitHub release draft for the tag that was pushed.
+  # It generates a changelog for the release and attaches build artifacts
+  # to the release as well. The release should be manually published on GitHub.
   release:
-    working_directory: /go/src/github.com/vapor-ware/synse-emulator-plugin
-    docker:
-      - image: circleci/golang:1.8
+    <<: *defaults
     steps:
+      - checkout
       - setup_remote_docker:
           docker_layer_caching: true
-      - checkout
       - run:
           name: Check Version matches Tag
-          command: |
-            version=$(make version)
-            if [ "${version}" != "${CIRCLE_TAG}" ]; then
-              echo "Version ${version} does not match Tag ${CIRCLE_TAG}"
-              exit 1
-            fi
+          command: make ci-check-version
       - run:
           name: Install Vendored Dependencies
-          command: |
-            make dep
-      - run:
-          name: Get Builder and Distributor
-          command: |
-            go get -v github.com/mitchellh/gox
-            go get -v github.com/tcnksm/ghr
+          command: make dep
       - run:
           name: Building Artifacts
-          command: |
-            gox --output="build/emulator_{{.OS}}_{{.Arch}}" -osarch='!darwin/386 !darwin/arm !darwin/arm64' -os='darwin linux'
+          command: make ci-build
       - run:
           name: Generate Changelog
           command: |
             docker pull timfallmk/github-changelog-generator
             docker run --name changelog timfallmk/github-changelog-generator \
-              -u vapor-ware \
-              -p synse-emulator-plugin \
-              --since-tag $(git describe --abbrev=0 --tags `git rev-list --tags --skip=1 --max-count=1`) \
-              -t ${GITHUB_TOKEN}
+              -u ${CIRCLE_PROJECT_USERNAME} \
+              -p ${CIRCLE_PROJECT_REPONAME} \
+              -t ${GITHUB_TOKEN} \
+              --since-tag $(git describe --abbrev=0 --tags `git rev-list --tags --skip=1 --max-count=1`)
             docker cp changelog:/usr/local/src/your-app/CHANGELOG.md ./
       - run:
           name: Create Release
           command: |
+            go get -v github.com/tcnksm/ghr
             if git describe --exact-match --tags HEAD; then
               CIRCLE_TAG=$(git describe --exact-match --tags HEAD)
             fi
             ghr \
-              -u ${GITHUB_USER} \
+              -u ${CIRCLE_PROJECT_USERNAME} \
```

Comment:
You _don't_ want all projects published under my name? How dare you!

Reply:
Oh. Maybe I'm just dumb. I think that I thought this was just the org name, e.g. vapor-ware.

Should probably change this back unless we actually have a "vapor-ware" user.. and if so, we'd need a token for it, right?

Reply:
we've got a `vaporauto` and `vaporio` user that could do this. OR to make things easier we could just stick with using your creds (: